go-controller/pkg/node/node.go

```diff
@@ -292,33 +292,44 @@ func (n *OvnNode) WatchEndpoints() error {
 			if reflect.DeepEqual(epNew.Subsets, epOld.Subsets) {
 				return
 			}
-			newEndpointAddressMap := buildEndpointAddressMap(epNew.Subsets)
-			for _, subset := range epOld.Subsets {
-				for _, address := range subset.Addresses {
-					if _, ok := newEndpointAddressMap[address.IP]; !ok {
-						deleteConntrack(address.IP)
-					}
+			newEpAddressMap := buildUDPEndpointAddressMap(epNew.Subsets)
+			for item := range buildUDPEndpointAddressMap(epOld.Subsets) {
+				if _, ok := newEpAddressMap[item]; !ok {
+					deleteConntrack(item.ip, item.port)
 				}
 			}
 		},
 		DeleteFunc: func(obj interface{}) {
 			ep := obj.(*kapi.Endpoints)
-			for _, subset := range ep.Subsets {
-				for _, address := range subset.Addresses {
-					deleteConntrack(address.IP)
-				}
+			epAddressMap := buildUDPEndpointAddressMap(ep.Subsets)
```

Comment:
@JacobTanenbaum could just do `for item := range buildUDPEndpointAddressMap(ep.Subsets) {`

Reply:
done